app/models/nic/base.rb

```diff
@@ -202,6 +204,19 @@ def sync_name
       end
     end
 
+    def validate_host_taxonomy
+      return true if self.subnet.nil?
+      return true if self.host.organization.nil? && self.subnet.organizations.empty? &&
+                     self.host.location.nil? && self.subnet.locations.empty?
+
+      errors.add(:subnet, _("is not defined for host\'s organization.")) unless include_or_empty?(self.subnet.organizations, self.host.organization)
```

Comment:
Logically it can apply to other properties: all sub-resources that are connected to the host should align with its taxonomy.
The case of NIC is a bit different though, because its taxonomy is indirect - it is hidden inside the subnet, and not directly on the NIC itself.
```
host(has org\loc) -> NIC -> subnet(taxable)
```